### ISSUE TITLE
separateLineBetweenGroups property has no effect

### ISSUE THREAD

**Issue by thomasturrell** [checkstyle/checkstyle#7537]
Check documentation: https://checkstyle.org/config_imports.html#CustomImportOrder

```
/var/tmp $ javac App.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8" />

  <property name="fileExtensions" value="java, properties, xml" />

  <module name="TreeWalker">
    <module name="CustomImportOrder">
      <property name="separateLineBetweenGroups" value="true" />
    </module>
  </module>

</module>

/var/tmp $ cat App.java
package com.demo.example;

import java.util.HashMap;
import java.util.Map;

import org.springframework.boot.autoconfigure.mail.MailProperties;

public class App 
{

}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.29-all.jar -c config.xml App.java
Starting audit...
[ERROR] App.java:6: Extra separation in import group before 'org.springframework.boot.autoconfigure.mail.MailProperties' [CustomImportOrder]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

Since separateLineBetweenGroups is true I did not expect any errors. Version 8.29 does not seem to respect the property separateLineBetweenGroups, Version 8.27 does not report any errors.

Version 8.27 output:

```
java $RUN_LOCALE -jar checkstyle-8.27-all.jar -c checkstyle.xml App.java
Starting audit...
Audit done.
```

--------------

**Comment by romani** [comment]
https://checkstyle.org/config_imports.html#CustomImportOrder_Properties

You are using default for `customImportOrderRules` property that is "all is one group". So behavior is correct. Please adjust such property to reference groups of imports.
Please beware of another Check https://checkstyle.org/config_imports.html#ImportOrder that might give different possibility to enforce order.

**Comment by thomasturrell** [comment]
Thank you @romani for taking the time to explain to me.

I guess I have a different definition of a group to Checkstyle. My idea of a group is com, org, etc. From the error output it wasn't obvious to me that my configuration would have no effect because of the `customImportOrderRules`.

It seems as though the behaviour has changed since 8.27. The following configuration behaves differently in 8.27 and 8.29.

```
    <module name="CustomImportOrder">
      <property name="sortImportsInGroupAlphabetically" value="true" />
      <property name="separateLineBetweenGroups" value="true" />
      <property name="customImportOrderRules" value="STATIC###THIRD_PARTY_PACKAGE" />
      <property name="tokens" value="IMPORT, STATIC_IMPORT, PACKAGE_DEF" />
    </module>
```

The above configuration is taken directly from the Google style guideline configuration provided by Checkstyle.

In order to upgrade checkstyle (as GItHub suggests I do) I have to reformat all of my code to remove separate lines between import groups.

My problem with reformatting is that I don't believe that the Eclipse formatter can be configured in a way to format the the imports inline with the Google style lines. In Eclipse I can either have spaces between groups or not and static imports are treated in the same way.  However I realise that not anything to do with checkstyle.

Thank you for your time, the discussion may be of value to anyone upgrading from Checkstyle 8.29 who uses the Google style guideline configuration provided by Checkstyle and Eclipse.

**Comment by romani** [comment]
>It seems as though the behaviour has changed since 8.27

Yes, it expected, you can see reference of such Check in release notes
https://checkstyle.org/releasenotes.html#Release_8.28

>My problem with reformatting is that I don't believe that the Eclipse formatter can be configured in a way to format the the imports inline with the Google style lines.

If that is not possible, so you should ask yourself a question "why your project need to follow Google way for imports order ?". Just match Eclipse setting to checkstyle and keep going. All you need is uniform order of imports to avoid extra diff.  Exact style of order is not really matter, engineers do not read imports 99% of time. 

>the discussion may be of value to anyone upgrading from Checkstyle 8.29 who uses the Google style guideline configuration provided by Checkstyle and Eclipse.

Sorry for problems. But each of our release find extra items that were not raised before.
I am not sure but I saw somewhere in web the Eclipse imports config to match their style.
If somebody know it, please share.
